src/structures/role.ts

```diff
@@ -2,10 +2,19 @@ import { Unpromise } from "../types/misc.ts";
 import { RoleData } from "../types/role.ts";
 
 export async function createRole(data: RoleData) {
+  const { tags, ...rest } = data;
+
+  const roleTags = {
+    botID: tags?.bot_id,
+    premiumSubscriber: "premium_subscriber" in (data.tags ?? {}),
```

Comment:
```suggestion
    premiumSubscriber: tags?.premium_subscriber,
```

Reply:
Why can't it be like this?

Reply:
The `tags.premium_subscriber` data sent by API may either exist with the value `null` or may not exist at all.

Reply:
If its just a boolean of true when it does exist, just put it inside a Boolean(tags?.premium_subscriber) that way for DD purposes it will ALWAYS be a boolean. true or false. We dont need to worry about optional or nullor anything

Reply:
Alright, sounds good!